client/core/core.go

```diff
@@ -2356,16 +2363,59 @@ func (c *Core) DiscoverAccount(dexAddr string, appPW []byte, certI interface{})
 		return dc.exchangeInfo(), false, nil
 	}
 
-	incompleteReg, paid, err := c.registerHD(dc, crypter)
+	// Setup our account keys and attempt to authorize with the DEX.
+	creds := c.creds()
+	encKey, _, err := dc.acct.setupCryptoV2(creds, crypter)
 	if err != nil {
-		return nil, false, err
+		return dc.exchangeInfo(), false, newError(acctKeyErr, "setupCryptoV2 error: %v", err)
+	}
+
+	// Instead of calling register, attempt authDEX.
+	err = c.authDEX(dc)
+	if err != nil {
+		var mErr *msgjson.Error
+		if errors.As(err, &mErr) && mErr.Code == msgjson.AccountNotFoundError {
+			return dc.exchangeInfo(), false, nil
+		}
+		return nil, false, newError(authErr, "unexpected authDEX error: %v", err)
+	}
+	if dc.acct.isSuspended {
+		c.log.Infof("HD account key for %s was reported as suspended. Register to make a new one.", dc.acct.host)
+		return dc.exchangeInfo(), false, nil // just paid=false, no error
+	}
+	// NOTE: An older server may not report account suspension status, just
+	// score. If the user discovers they cannot place orders (and their score is
+	// bad), they should disable the account and call Register directly.
```

Comment:
Is it possible to detect from the place order error message that the account is suspended?

Reply:
Programatically... not reliably.  The server returns an error coded somewhat vaguely with `MarketNotRunningError` for that.

https://github.com/decred/dcrdex/blob/5e2ab91533d3fdf9ce774d104fbff844347baa1f/server/market/orderrouter.go#L215-L216

But the user does see a clear message in red *on the order confirmation form*, so yes.

Reply:
Regarding this note you're commenting on ("An older server may not report account suspension status"), to be fair, and older server won't be reporting the helpful error codes and data we expect from the register route either, since the initial HD keys PR modified the register request handler. https://github.com/decred/dcrdex/commit/b66a35f24215456763a16cee9c32875f61f9814b#diff-e7126c606b49bf8ca255171ae2f485e8d92f442af17e953700f49bcea7215507R62-R77
So it would seem to be no less reliable.

Reply:
This means #1204 will also have to deal with (skip) HD accounts that are manually disabled if the user wants to re-register with the DEX.

Reply:
Re #1204, do you want this to go in first before a fix for #1204?

Reply:
I'm not even sure about #1204 for 0.3 or not.
This PR certainly doesn't depend on it, at least there are no regressions with this PR compared to master.  So I think this should go in regardless.

There certainly are a lot of edge cases. I'm wondering if manually disabling an HD account should simply be disallowed.  If it is suspended, the software can disable it.  That way suspended <==> disabled for HD accounts.

Reply:
If an HD account is suspended by a legacy server, can the software determine the suspension and thus auto-disable the account?

#1024 doesn't seem to me to be major work but important - prevent users from creating new DEX accounts with unrecoverable keys once they've started using a seed. Might be a hard to hit edge, but it wouldn't make sense to have and use a seed if a DEX account still cannot be recovered in certain situations.

Reply:
> If an HD account is suspended by a legacy server, can the software determine the suspension and thus auto-disable the account?

Seems like the answer is no, not really.

However, although a 0.2 patch release added the dex pub key to the config response, thus enabling HD accounts with that, it's not actually deployed anywhere, and doesn't need to be if we are now realizing that's creating a pickle for us.

It's also not unreasonable to require the server to be upgraded for 0.3 clients if we want to go that route.

Reply:
Just remembered that legacy keys will still be used to support older servers. Did we agree to NOT do that; and compel clients to only register with servers that support HD account keys going forward?

Reply:
No, I'm just saying that's on the table if it untangles this mess most efficiently.  I'm not sure yet.

Reply:
Should we move on to address these questions in https://github.com/decred/dcrdex/pull/1210 then?

Reply:
> Just remembered that legacy keys will still be used to support older servers.

And imported accounts.